routing.asciidoc

```diff
@@ -1,160 +1,218 @@
+We call this deadline a "time lock". 
+The deposit is locked with the escrow service for a certain amount of time, and is eventually released even if no proof of payment was provided.
+
+In order to factor this in, the contract between Alice and Bob is once again amended with a new clause:
 
-_Bob has 24 hours to show the secret after the contract was signed. If the time has passed Alice will get her deposit back from the escrow service and the contract becomes invalid._
+[alice-gloria-routing-7]
+----
+_Bob has 24 hours to show the secret after the contract was signed.
+If he does not provide the secret by this time, Alice's deposit will be refunded by the escrow service and the contract becomes invalid._
+----
 
-Bob of course now has to make sure to receive the proof of payment before 24 hours is up.
-If he would receive this proof after 24 hours, the contract between him and Alice would already be nullified and Bob would not get reimbursed.
-After he signed the contract with Alice he creates a new (altered) original contract between him and Wei that reads:
+Bob, of course, now has to make sure he receives the proof of payment within 24 hours.
+Even if he successfully pays Wei, if he receives the proof of payment later than 24 hours he will not be reimbursed.
+In turn, he will alter his contract with Wei in the following way:
 
-_Wei has 22 hours to show the secret after the contract was signed. If the time has passed Bob will get his deposit back from the escrow service and the contract becomes invalid._
+[alice-gloria-routing-8]
+----
+_Wei has 22 hours to show the secret after the contract was signed.
+If he does not provide the secret by this time, Bob's deposit will be refunded by the escrow service and the contract becomes invalid._
+----
 
-As you have guessed Wei is now incentivised create a _new_ (altered) contract with Gloria:
+	As you might have guessed, Wei is now incentiviced to also alter his contract with Gloria:
 
-_Gloria has 20 hours to show the secret after the contract was signed. If the time has passed Wei will get his deposit back from the escrow service and the contract becomes invalid._
+[alice-gloria-routing-9]
+----
+_Gloria has 20 hours to show the secret after the contract was signed.
+If he does not provide the secret by this time, Bob's deposit will be refunded by the escrow service and the contract becomes invalid._
+----
 
-With such a chain of contracts we can be sure that after 24 hours of setting up the first contract that the payment was either successfully delivered from Alice via Bob and Wei to Gloria or that the payment has failed and was not conducted at all.
+With such a chain of contracts we can ensure that, after 24 hours, the payment will succesfully deliver from Alice to Bob to Wei to Gloria, or it will fail and everyone will be refunded.
```

Comment:
This part I am not sure about. Consider the corner case:

Gloria reveals the secret to Wei, gets 10 gold coins
Wei reveals the secret to Bob, gets 11 gold coins
Bob's node crashes, goes offline, and he cannot reveal the secret to Alice by the time the HTLC expires.
The HTLC expires, the escrow ends, Alice is refunded her 12 gold coins.

Gloria ends up with 10, Wei ends up with 1, and Alice still has her 12 coins.
Bob paid Wei 11, but did not receive 12 from Alice so he's lost 11 coins and essentially picked up the tab.
What now?